### PR TITLE
`occ_impl` doc strings

### PR DESCRIPTION
**Pull request by marcus7070**
I've realised that there are a few useful methods in the `occ_impl` layer that I wasn't familar with. So I'm going through and adding docstrings to a few and fixing any inconsistent formatting or other small things I see.

### CODE REVIEW

**Comment by marcus7070** [comment]
Before I get too far I thought I should just check, these methods don't have doc strings because no one has got around to it yet, correct? Not because they are supposed to be hidden from users or sphinx? Or because the OCCT docs are supposed to be the only source and we don't want to repeat this stuff in CQ?

**Comment by marcus7070** [comment]
I tried to squash commits and something went horribly wrong. Successfully squashed with that second force push now.

**Comment by marcus7070** [comment]
There might be a problem with sphinx auto type hints. When `self` is type hinted it does this:
screenshot2020-10-24-173549
Is that correct behaviour?

**Comment by marcus7070** [comment]
I'm done for now. If I'm feeling masochistic again I'll try to do some more in another PR.

**Comment by adam-urbanczyk** [comment]
Thanks @marcus7070  - I think it was due to all of the above reasons.

**Comment by marcus7070** [comment]
> I think it was due to all of the above reasons.

Ok then, if there are reasons to not have doc strings on any of these methods then please do remove them from this PR or reject it entirely. I don't mind; I did the work to familiarise myself with the methods, it's not wasted effort.

**Review by adam-urbanczyk — APPROVED** [review]

**Comment by adam-urbanczyk** [comment]
Thanks @marcus7070 , I think we should merge it before 2.1. In the end it does not hurt to have better docstrings.

File: cadquery/occ_impl/shapes.py [hunk 1]
```diff
@@ -677,16 +779,19 @@ def transformShape(self, tMatrix: Matrix) -> "Shape":
...
-            WARNING: transformGeometry will sometimes convert lines and circles to splines,
-            but it also has the ability to handle skew and stretching transformations.
+        If your transformation is only translation and rotation, it is safer to
+        use :py:meth:`transformShape`, which doesnt change the underlying type
+        of the geometry, but cannot handle skew transformations.
 
-            If your transformation is only translation and rotation, it is safer to use transformShape,
-            which doesnt change the underlying type of the geometry, but cannot handle skew transformations
+        :param tMatrix: The transformation matrix
+        :returns: a copy of the object, but with geometry transformed insetad
```
Comment (jmwright) [inline cadquery/occ_impl/shapes.py:793]: instead is misspelled here

**Review by jmwright — APPROVED** [review]
Other than my comment, this is ready to merge. Thanks @marcus7070

File: cadquery/occ_impl/shapes.py (hunk 1 above)
Comment (marcus7070) [reply cadquery/occ_impl/shapes.py]: Fixed, thanks.

**Comment by adam-urbanczyk** [comment]
Merging, thanks @marcus7070 .

**Pull request merged**